lntest/wait/wait.go

```diff
@@ -58,6 +58,13 @@ func NoError(f func() error, timeout time.Duration) error {
 	// If f() doesn't succeed within the timeout, return the last
 	// encountered error.
 	if err := Predicate(pred, timeout); err != nil {
+		// Handle the case where the passed in method, f, hangs for the
```

Comment:
I think `Predicate` already handles this and returns an error `fmt.Errorf("predicate not satisfied after...` after the timeout?

Reply:
yes but that error is not the one being returned

Reply:
oh, I see, you mean I should just return `err` here? sounds good :)

Reply:
ah - I think my original reason for making a custom message here is that if a user is using `NoError` then `predictae not satisfied after timeout` doesnt realllly tell them that the function they are testing is just not returning

Reply:
let me know what you think :)

Reply:
yeah I see what you mean, `predicate not satisfied after...` doesn't tell us much. We could change the error there, or the way you do it now, no preference:)

I think we'll make the `Predicate` method private, and force the usage of `wait.NoError` only. Then we can make `Predicate` return the `f`'s error by `result := make(chan error, 1)`, thus getting rid of `predErr`. Maybe another PR tho.